frame/support/src/traits/tokens/nonfungibles_v2.rs

```diff
@@ -0,0 +1,243 @@
+		None
+	}
+
+	/// Returns the attribute value of `item` of `collection` corresponding to `key`.
+	///
+	/// By default this is `None`; no attributes are defined.
+	fn attribute(
+		_collection: &Self::CollectionId,
+		_item: &Self::ItemId,
+		_key: &[u8],
+	) -> Option<Vec<u8>> {
+		None
+	}
+
+	/// Returns the strongly-typed attribute value of `item` of `collection` corresponding to
+	/// `key`.
+	///
+	/// By default this just attempts to use `attribute`.
+	fn typed_attribute<K: Encode, V: Decode>(
+		collection: &Self::CollectionId,
+		item: &Self::ItemId,
+		key: &K,
+	) -> Option<V> {
+		key.using_encoded(|d| Self::attribute(collection, item, d))
+			.and_then(|v| V::decode(&mut &v[..]).ok())
+	}
+
+	/// Returns the attribute value of `collection` corresponding to `key`.
+	///
+	/// By default this is `None`; no attributes are defined.
+	fn collection_attribute(_collection: &Self::CollectionId, _key: &[u8]) -> Option<Vec<u8>> {
+		None
+	}
+
+	/// Returns the strongly-typed attribute value of `collection` corresponding to `key`.
+	///
+	/// By default this just attempts to use `collection_attribute`.
+	fn typed_collection_attribute<K: Encode, V: Decode>(
+		collection: &Self::CollectionId,
+		key: &K,
+	) -> Option<V> {
+		key.using_encoded(|d| Self::collection_attribute(collection, d))
+			.and_then(|v| V::decode(&mut &v[..]).ok())
+	}
+
+	/// Returns `true` if the `item` of `collection` may be transferred.
+	///
+	/// Default implementation is that all items are transferable.
+	fn can_transfer(_collection: &Self::CollectionId, _item: &Self::ItemId) -> bool {
+		true
+	}
+}
+
+/// Interface for enumerating items in existence or owned by a given account over many collections
+/// of NFTs.
+pub trait InspectEnumerable<AccountId>: Inspect<AccountId> {
+	/// Returns an iterator of the collections in existence.
+	fn collections() -> Box<dyn Iterator<Item = Self::CollectionId>>;
+
+	/// Returns an iterator of the items of a `collection` in existence.
+	fn items(collection: &Self::CollectionId) -> Box<dyn Iterator<Item = Self::ItemId>>;
+
+	/// Returns an iterator of the items of all collections owned by `who`.
+	fn owned(who: &AccountId) -> Box<dyn Iterator<Item = (Self::CollectionId, Self::ItemId)>>;
+
+	/// Returns an iterator of the items of `collection` owned by `who`.
+	fn owned_in_collection(
+		collection: &Self::CollectionId,
+		who: &AccountId,
+	) -> Box<dyn Iterator<Item = Self::ItemId>>;
+}
+
+/// Trait for providing the ability to create collections of nonfungible items.
+pub trait Create<AccountId, CollectionConfig>: Inspect<AccountId> {
+	/// Create a `collection` of nonfungible items to be owned by `who` and managed by `admin`.
+	fn create_collection(
+		collection: &Self::CollectionId,
+		who: &AccountId,
+		admin: &AccountId,
+		config: &CollectionConfig,
+	) -> DispatchResult;
+}
+
+/// Trait for providing the ability to destroy collections of nonfungible items.
+pub trait Destroy<AccountId>: Inspect<AccountId> {
```

Comment:
Why does a `Destroy` need to be an `Inspect`?

Reply:
Idk, it's a copy of the existing `frame/support/src/traits/tokens/nonfungibles.rs` I only added support for feature flags